### ISSUE TITLE
Rustc crashes with nul byte inside inline assembly

### ISSUE THREAD

**Issue by andrewreds** [rust-lang/rust#69092]
#### Fails
```rust
#![feature(asm)]

fn main() {
    unsafe { asm!(".ascii \"Xen\0\""); }
}
```

#### Works
```rust
#![feature(asm)]

fn main() {
    unsafe { asm!(".ascii \"Xen\\0\""); }
}
```

```rust
#![feature(asm)]

fn main() {
    unsafe { asm!(".asciz \"Xen\""); }
}
```

### Meta
`rustc --version --verbose`:
```
rustc 1.43.0-nightly (fc23a8183 2020-02-11)
binary: rustc
commit-hash: fc23a81831d5b41510d3261c20c34dd8d32f0f31
commit-date: 2020-02-11
host: x86_64-unknown-linux-gnu
release: 1.43.0-nightly
LLVM version: 9.0
```

### Error output

```
thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: NulError(11, [46, 97, 115, 99, 105, 105, 32, 34, 88, 101, 110, 0, 34])', src/librustc_codegen_llvm/asm.rs:77:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.43.0-nightly (fc23a8183 2020-02-11) running on x86_64-unknown-linux-gnu

```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1052
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1428
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: rustc_driver::report_ice
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
  12: rust_begin_unwind
             at src/libstd/panicking.rs:378
  13: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  14: core::option::expect_none_failed
             at src/libcore/option.rs:1211
  15: rustc_codegen_llvm::asm::<impl rustc_codegen_ssa::traits::asm::AsmBuilderMethods for rustc_codegen_llvm::builder::Builder>::codegen_inline_asm
  16: rustc_codegen_ssa::mir::block::<impl rustc_codegen_ssa::mir::FunctionCx<Bx>>::codegen_block
  17: rustc_codegen_ssa::mir::codegen_mir
  18: <rustc::mir::mono::MonoItem as rustc_codegen_ssa::mono_item::MonoItemExt>::define
  19: rustc_codegen_llvm::base::compile_codegen_unit::module_codegen
  20: rustc::dep_graph::graph::DepGraph::with_task
  21: rustc_codegen_llvm::base::compile_codegen_unit
  22: rustc_codegen_ssa::base::codegen_crate
  23: <rustc_codegen_llvm::LlvmCodegenBackend as rustc_codegen_utils::codegen_backend::CodegenBackend>::codegen_crate
  24: rustc_interface::passes::QueryContext::enter
  25: rustc_interface::queries::Queries::ongoing_codegen
  26: rustc_interface::interface::run_compiler_in_existing_thread_pool
  27: scoped_tls::ScopedKey<T>::set
  28: syntax::attr::with_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</p>
</details>

This issue has been assigned to @euclio via [this comment](https://github.com/rust-lang/rust/issues/69092#issuecomment-585277602).

**Comment by euclio** [comment]
I'd like to work on this.

@rustbot claim

**Comment by JohnTitor** [comment]
Now ICE no longer occurs, I think it's fixed by #69893. Marked as E-needstest.
